scripts/pulling_sim4life_images/pull_images.py

```diff
@@ -0,0 +1,180 @@
+"tip" : {"key": "osparc-infra/docs/deployments-hardware-and-credentials/tip.itis.swiss/osparc-public.pem", "e2e-name" : "tip.itis.swiss", "sparc-external-name": "tip-public", "registry" : "registry.tip.itis.swiss" },
+"osparc-staging.speag.com" : {"key": "osparc-infra/docs/deployments-hardware-and-credentials/osparc.speag.com/osparc-speag.pem", "e2e-name" : "Dalco", "sparc-external-name": "dalco-staging-production", "registry" : "registry.osparc-staging.speag.com" },
+"osparc.speag.com" : {"key": "osparc-infra/docs/deployments-hardware-and-credentials/osparc.speag.com/osparc-speag.pem", "e2e-name" : "Dalco", "sparc-external-name": "dalco-staging-production", "registry" : "registry.osparc.speag.com" },}
+
+args = ["master", "aws-production", "aws-staging", "dalco-staging", "dalco-production", "tip"]
+
+
+# This function parse yaml data from e2e-ops and return a list of hosts to ssh
+def yaml_data_to_hosts(stack_arg):
+    global stacks
+    stack = stacks[stack_arg]
+    with open("e2e-ops/data/machineCredentials.yml", 'r') as file:
+        yaml_data = yaml.load(file, Loader=yaml.FullLoader)
+        clusters = yaml_data["Clusters"]
+        test_occurrence = next((item for item in clusters if stack["e2e-name"] in item), None)
+        hosts_yaml = test_occurrence[stack["e2e-name"]]
+        hosts = []
+        for host in hosts_yaml:
+            for ho in host.values():
+                hosts.append(ho)
+    return hosts
+
+
+# This function parse yaml data from sparc-external file and return the versions of sim4life that needs to be pulled.
+def yaml_data_to_pulling_info(stack_arg):
+    global stacks
+    stack = stacks[stack_arg]
+    pull_images = []
+    images_names = ["ci/osparc-osparc-s4l/master/s4l-core-lite", "ci/osparc-osparc-s4l/master/sym-server-dy"] 
+    with open("sparc-external/sync-workflow.yml", 'r') as file:
+        yaml_data = yaml.load(file, Loader=yaml.FullLoader)
+        images = yaml_data["stages"]
+        for sim4life_image in images_names:
+            for image in images:
+                if image["from"]["repository"] == sim4life_image:
+                    for destinations in image["to"]:
+                        if destinations["destination"] == stack["sparc-external-name"]:
+                            tags = len(destinations["tags"])
+                            tag = destinations["tags"][tags - 1]
+                            pull_images.append(destinations["repository"] + ":" + tag)
+    return pull_images
+
+
+def ssh(hosts, images_to_pull, stack_arg):
+    for host in hosts:
+        host_ip = host["IP"]
+        port = 22
+        if "SSHPort" in host :
+            port = host["SSHPort"]
+        
+        username = host["Username"]
+        global stacks
+        stack = stacks[stack_arg]
+        subprocess.run("chmod 600 " + stack["key"], shell=True)
```

Comment:
shell-True is considered bad practice as it can be exploited. Just for info, for this specific script it should be fine.